### ISSUE TITLE
Steam crashes immediately after startup

### ISSUE THREAD

**Issue by peti** [NixOS/nixpkgs#91349]
I'm using a current version of NixOS `master` @ 740fa4dc5b326f98a5e7cc5d0b590e9803287889. I've installed `Steam` and tried to run it, but I get an immediate crash:

~~~
~ $ steam
** (process:23130): ERROR **: 11:16:15.235: bind_mount: mount(source, target, NULL, MS_BIND | MS_REC, NULL): No such device
zsh: trace trap (core dumped)  steam
~~~

The relevant bit of the strace output is:

~~~
mkdir("/tmp/chrootenv1IKXM0", 0755)     = 0
clone(child_stack=NULL, flags=CLONE_CHILD_CLEARTID|CLONE_CHILD_SETTID|SIGCHLD, child_tidptr=0x7fcf9d97d0d0) = 23751
wait4(23751, [{WIFSIGNALED(s) && WTERMSIG(s) == SIGTRAP && WCOREDUMP(s)}], 0, NULL) = 23751
--- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_DUMPED, si_pid=23751, si_uid=1000, si_status=SIGTRAP, si_utime=0, si_stime=0} ---
rmdir("/tmp/chrootenv1IKXM0")           = 0
getpid()                                = 23749
kill(23749, SIGTRAP)                    = 0
--- SIGTRAP {si_signo=SIGTRAP, si_code=SI_USER, si_pid=23749, si_uid=1000} ---
+++ killed by SIGTRAP (core dumped) +++
~~~

I have installed the 32-bit versions of the OpenGL libraries etc. and I've even had Steam running at some point, but now it's stopped working for no apparent reason. Rebooting the machine didn't help either.

**Comment by peti** [comment]
Here is another strace that looks slightly different:

~~~
mkdir("/tmp/chrootenvEP5OM0", 0755)     = 0
clone(child_stack=NULL, flags=CLONE_CHILD_CLEARTID|CLONE_CHILD_SETTID|SIGCHLD, child_tidptr=0x7f4daa16e0d0) = 5192
wait4(5192, 
** (process:5192): ERROR **: 11:25:55.219: bind_mount: mount(source, target, NULL, MS_BIND | MS_REC, NULL): No such device
[{WIFSIGNALED(s) && WTERMSIG(s) == SIGTRAP && WCOREDUMP(s)}], 0, NULL) = 5192
--- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_DUMPED, si_pid=5192, si_uid=1000, si_status=SIGTRAP, si_utime=0, si_stime=0} ---
rmdir("/tmp/chrootenvEP5OM0")           = 0
getpid()                                = 5189
kill(5189, SIGTRAP)                     = 0
--- SIGTRAP {si_signo=SIGTRAP, si_code=SI_USER, si_pid=5189, si_uid=1000} ---
+++ killed by SIGTRAP (core dumped) +++
zsh: trace trap (core dumped)  strace steam
~~~

Note the "no such device" error. That looks interesting?

**Comment by pbogdan** [comment]
On an off chance is there anything interesting if you run strace to include the child processes as well?

**Comment by peti** [comment]
I found out what's causing these crashes. My machine has a couple of auto-mounted NFS shares configured in `/etc/fstab`, e.g.: 

    foo.example.org:/bar /bar nfs4 noauto,ro,x-systemd.automount,x-systemd.idle-timeout=5min,_netdev 0 0

Now, the servers that provide those mount points are reachable only after I've started a VPN. When I run `steam` with that VPN up, it works fine. If I run `steam` with the VPN down, it crashes.

**Comment by austinbutler** [comment]
To give more data I have a similar setup but I do not have any issues running Steam even when this NFS server is offline.

```
example.net:/mnt/data /mnt/nas nfs soft,noauto,noatime,nodiratime,x-systemd.automount,x-systemd.mount-timeout=10,timeo=14,users 0 0
```

**Comment by ikervagyok** [comment]
I have `XMonad` and `WireGuard` over WiFi on my laptop and am seeing the same issue. The interesting part is, that whenever steam gets past the login/loading screen the whole X11 session freezes in a way, that I can't even switch to the TTYs. But pressing the power-button works just fine. I don't have a second machine at hand, to test more thoroughly over SSH though.

**Comment by jonringer** [comment]
steam has been moved over to using `buildFHSUserEnvBubblewrap`, and I have haven't see this issue with it.

Please open a new ticket if it's still broken.
